pkg/trustedresources/verify.go

```diff
@@ -183,18 +199,34 @@ func verifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.
 				break
 			}
 		}
-		// if this policy fails the verification and the mode is not "warn", should return error directly. No need to check other policies
 		if !passVerification {
-			if p.Spec.Mode == v1alpha1.ModeWarn {
-				logger := logging.FromContext(ctx)
-				logger.Warnf("%w: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())
-			} else {
-				// if the mode is "enforce" or not set, return error.
-				return fmt.Errorf("%w: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())
+			return &VerificationResult{VerificationResultType: VerificationError, Err: fmt.Errorf("%w: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())}
+		}
+	}
+
+	// return VerificationWarn type of VerificationResult if any warn policies fails.
+	for _, p := range warnPolicies {
+		passVerification := false
+		verifiers, err := verifier.FromPolicy(ctx, k8s, p)
+		if err != nil {
+			warn := fmt.Errorf("fails to get verifiers for resource %s from namespace %s: %w", resource.GetName(), resource.GetNamespace(), err)
+			logger.Warnf(warn.Error())
+			return &VerificationResult{VerificationResultType: VerificationWarn, Err: warn}
+		}
+		for _, verifier := range verifiers {
```

Comment:
would it make sense to break this into a helper function?

Reply:
I was referring to the verifier loop. I still think it makes sense, since e.g. what if we decided to change the logic to "and" instead of "or"?

Reply:
sure! added a helper function